src/java.base/share/classes/jdk/internal/foreign/abi/Binding.java

```diff
@@ -202,7 +203,7 @@ void interpret(Deque<Object> stack, StoreFunc storeFunc,
                    LoadFunc loadFunc, SegmentAllocator allocator);
 
     private static void checkType(Class<?> type) {
```

Comment:
For another day: I wonder if we might be better served by more specific "check" routines. E.g. I suppose `Object` is not valid always... and maybe in other cases only certain primitives are ok.